src/components/offers-sorting/offers-sorting.jsx

```diff
@@ -1,21 +1,55 @@
-import React from 'react';
+import React, {useState, useEffect, createRef} from 'react';
+
+import {SortingTypes} from '../../const';
+
+import classnames from 'classnames';
 
 const OffersSorting = () => {
+  const [opened, setOpened] = useState(false);
+  const [active, setActive] = useState(SortingTypes[0]);
+  const formNode = createRef();
+
+  const handleDropdownClick = () => {
+    setOpened((prevState) => !prevState);
+  };
+
+  const handleItemClick = (evt) => {
+    setActive(evt.target.innerText);
+    setOpened(false);
+  };
+
+  const handleClickOutside = (evt) => {
+    if (formNode.current.contains(evt.target)) {
+      return;
+    }
+    setOpened(false);
+  };
+
+  useEffect(() => {
+    if (opened) {
+      document.addEventListener(`mousedown`, handleClickOutside);
+    } else {
+      document.removeEventListener(`mousedown`, handleClickOutside);
+    }
```

Comment:
```suggestion
    document[opened ? 'addEventListener' : 'removeEventListener'](`mousedown`, handleClickOutside);
```
as an option, just for fun :)